src/plugins/better-navigation/index.ts

```diff
@@ -0,0 +1,179 @@
+import { hookIntoOverrideKeystroke } from "src/utils/listenerHelpers";
+import { MathQuillField, MathQuillView } from "src/components";
+import { getController } from "../intellisense/latex-parsing";
+
+import "./index.less";
+
+function isSupSubscriptMQElem(el?: HTMLElement) {
+  return el?.classList.contains("dcg-mq-supsub");
+}
+
+function isWordMQElem(el?: HTMLElement) {
+  return (
+    el &&
+    (el.classList.contains("dcg-mq-digit") ||
+      el.tagName.toUpperCase() === "VAR" ||
+      el.classList.contains("dcg-mq-supsub"))
+  );
+}
+
+function isCtrlArrowSkippableSymbolMQElem(el?: HTMLElement) {
+  return (
+    el?.classList.contains("dcg-mq-bracket-container") ||
+    el?.classList.contains("dcg-mq-fraction") ||
+    el?.classList.contains("dcg-mq-large-operator") ||
+    el?.classList.contains("dcg-mq-int") ||
+    el?.classList.contains("dcg-mq-sqrt-container") ||
+    el?.classList.contains("dcg-mq-nthroot-container")
+  );
+}
+
+function isAtStartOrEndOfASubscriptOrSuperscript(
+  mq: MathQuillField,
+  right: boolean
+) {
+  const ctrlr = getController(mq);
+  return (
+    (ctrlr.cursor?.parent?._el?.classList.contains("dcg-mq-sup") ||
+      ctrlr.cursor?.parent?._el?.classList.contains("dcg-mq-sub")) &&
+    !ctrlr.cursor?.[right ? 1 : -1]
+  );
+}
+
+interface BetterNavSettings {
+  ctrlArrow: boolean;
+  scrollableExpressions: boolean;
+}
+
+export default class BetterNavigation extends PluginController<BetterNavSettings> {
+  static id = "better-navigation" as const;
+  static enabledByDefault = true;
+  static config = [
+    {
+      type: "boolean",
+      default: true,
+      key: "ctrlArrow",
+    },
+    {
+      type: "boolean",
+      default: false,
+      key: "scrollableExpressions",
+    },
+  ] as const;
+
+  removeHandlers() {
+    for (const handler of this.customRemoveHandlers) handler();
+  }
+
+  afterConfigChange(): void {
+    document.body.classList.toggle(
+      "dsm-better-nav-scrollable-expressions",
+      this.settings.scrollableExpressions
+    );
+    this.removeHandlers();
+  }
+
+  dispatcherID: string | undefined;
+
+  customRemoveHandlers: (() => void)[] = [];
+
+  keydownHandler = () => {
+    if (Calc.focusedMathQuill && this.settings.ctrlArrow) {
+      const remove = hookIntoOverrideKeystroke(
+        Calc.focusedMathQuill.mq,
+        (key, _) => {
+          const mq = MathQuillView.getFocusedMathquill();
+
+          if (!mq) return true;
+          if (
+            key !== "Ctrl-Left" &&
+            key !== "Ctrl-Right" &&
+            key !== "Ctrl-Shift-Left" &&
+            key !== "Ctrl-Shift-Right" &&
+            key !== "Ctrl-Backspace"
+          )
+            return true;
+
+          // backspace is implicitly "left"
+          const right = key === "Ctrl-Right" || key === "Ctrl-Shift-Right";
+          const shift = key === "Ctrl-Shift-Left" || key === "Ctrl-Shift-Right";
+          const backspace = key === "Ctrl-Backspace";
```

Comment:
It's easy to get the logic wrong here. I would write it as a mapping from key to attributes, like

```
{
  "Ctrl-Left": {dir: -1, mode: "move"},
  "Ctrl-Right": {dir: +1, mode: "move"},
  "Ctrl-Shift-Left": {dir: -1, mode: "extend-sel"},
  "Ctrl-Shift-Right": {dir: +1, mode: "extend-sel"},
  "Ctrl-Backspace": {dir: -1, mode: "delete"},
  "Ctrl-Delete": {dir: +1, mode: "delete"},
  "Ctrl-Shift-Backspace": {dir: -1, mode: "delete"},
  "Ctrl-Shift-Delete": {dir: +1, mode: "delete"},
}
```

(not sure about `Ctrl-Shift-Backspace` and `Ctrl-Shift-Delete` since they're the same as the non-shift forms, but I can't think of what shift would do otherwise)

Then checking for presence in the mapping also gives you the direction and action to take.

(Note I added "Delete" to this map -- will have to update `arrowOp` to match, or put it in the mapping as well).

Reply:
I like this approach. Not totally sure if we need `Ctrl+Shift+Backspace` and `Ctrl+Shift+Delete` (from a bit of brief testing I've found text editors tend to do nothing with them) so it might be worth keeping them unused on the miniscule offchance that we need them for something else lol

Reply:
I've implemented this, just without the ctrl-shift-backspace/delete. Btw they call delete `Del` in MQ, not `Delete`.

Reply:
nice